tests/php/test_class.jetpack.php

```diff
@@ -321,4 +321,64 @@ public function test_dns_prefetch() {
 
 		$this->assertEquals( $expected, get_echo( array( 'Jetpack', 'dns_prefetch' ) ) );
 	}
+
+	/**
+	 * @author jubstuff
+	 * @covers Jetpack::maybe_min_assets
+	 * @since 4.1.0
+	 */
+	function test_maybe_min_assets() {
+		add_filter( 'jetpack_load_minified_scripts', '__return_true' );
+
+		$scripts = array(
+			'script1.min.js',
+			'my.js.script6.min.js',
+			'script7.js.min.js',
+			'case-sensitive.min.js',
+			'core.jsdom.min.js'
+		);
+
+		$base      = dirname( plugin_basename( JETPACK__PLUGIN_FILE ) );
+		$base_path = JETPACK__PLUGIN_DIR . substr( $base, strlen( basename( JETPACK__PLUGIN_DIR ) ) + 1 ) . '/_inc';
+
+		if( ! is_dir( $base_path ) ) {
+			mkdir( $base_path, 0777, true );
+		}
+		foreach ( $scripts as $script ) {
+			@touch( "$base_path/$script" );
```

Comment:
The `Jetpack::maybe_min_asset` method checks for file existence before returning the minified file URL (which I think is correct). 
I thought about splitting the method, but since there were no tests, I wouldn't know if I had introduced regressions.

Do you think it is better for tests not to use the filesystem?